File: win/directx/sdk-9.0/Bin/DXUtils/Visual Studio 7.0 Wizards/VB .NET/VBWizards/Mdx9AppWizard/Templates/1033/graphics.vb
Imports System
Imports System.Windows.Forms
Imports System.Drawing

'/ <summary>
'/ This class is where the Drawing routines
'/ for non-Direct3D and non-DirectDraw 
'/ applications reside.
'/ </summary>
Public Class GraphicsClass
    Private spriteSize As Single = 50
    Private owner As Control = Nothing
    Private graphicsWindow As Graphics = Nothing


    Public Sub New(ByVal owner As Control)
        Me.owner = owner
    End Sub 'New


    Public Sub RenderGraphics(ByVal destination As Point)
        If Not owner.Created Then
            Return
        End If
        graphicsWindow = owner.CreateGraphics()
        graphicsWindow.Clear(Color.Blue)
        graphicsWindow.FillEllipse(Brushes.Black, destination.X, destination.Y, spriteSize, spriteSize)
    End Sub 'RenderGraphics
End Class 'GraphicsClass

File: win/directx/sdk-9.0/Bin/DXUtils/Visual Studio 7.0 Wizards/C#/VC#Wizards/Mdx9AppWizard/Templates/1033/graphics.cs
using System;
using System.Windows.Forms;
using System.Drawing;

/// <summary>
/// This class is where the Drawing routines
/// for non-Direct3D and non-DirectDraw 
/// applications reside.
/// </summary>
public class GraphicsClass
{
    private const float spriteSize = 50;
    private Control owner = null;
    private Graphics graphicsWindow = null;
    
    public GraphicsClass(Control owner)
    {
        this.owner = owner;
    }

    public void RenderGraphics(Point destination)
    {
        if (!owner.Created)
            return;

        graphicsWindow = owner.CreateGraphics();
        graphicsWindow.Clear(Color.Blue);
        graphicsWindow.FillEllipse(Brushes.Black, destination.X, destination.Y, spriteSize, spriteSize);
    }
}